polaris-react/src/components/Grid/Grid.tsx

```diff
@@ -0,0 +1,110 @@
+import React, {useState, useMemo} from 'react';
+
+import {debounce} from '../../utilities/debounce';
+import breakpoints from '../../tokens/token-groups/breakpoints.json';
+import {useEventListener} from '../../utilities/use-event-listener';
+
+import {Cell} from './components';
+import styles from './Grid.scss';
+
+type Breakpoints =
+  keyof typeof breakpoints extends `breakpoints-${infer Breakpoint}`
+    ? Breakpoint
+    : never;
+
+type Columns = {
+  [Breakpoint in Breakpoints]?: number;
+};
+
+type Areas = {
+  [Breakpoint in Breakpoints]?: string[];
+};
+
+type Gap = {
+  [Breakpoint in Breakpoints]?: string;
+};
+
+export interface GridProps {
+  /* Grid gap */
+  gap?: Gap;
+  /* Set grid-template-areas */
+  areas?: Areas;
+  /* Must be less than 12 to work with the grid system */
+  columns?: Columns;
+  children?: React.ReactNode;
+}
+/** **Experimental!** This component is in alpha. Use with caution. */
+export const Grid: React.FunctionComponent<GridProps> & {
+  Cell: typeof Cell;
+} = function Grid({gap, areas, children, columns}: GridProps) {
+  const [gridTemplateAreas, setGridTemplateAreas] = useState(getAreas(areas));
+  const style = {
+    '--pc-grid-gap-xs': gap?.xs,
+    '--pc-grid-gap-sm': gap?.sm,
+    '--pc-grid-gap-md': gap?.md,
+    '--pc-grid-gap-lg': gap?.lg,
+    '--pc-grid-gap-xl': gap?.xl,
+    '--pc-grid-columns-xs': columns?.xs,
+    '--pc-grid-columns-sm': columns?.sm,
+    '--pc-grid-columns-md': columns?.md,
+    '--pc-grid-columns-lg': columns?.lg,
+    '--pc-grid-columns-xl': columns?.xl,
```

Comment:
```suggestion
    ...(gap?.xs && {'--pc-grid-gap-xs': gap?.xs}),
    ...(gap?.sm && {'--pc-grid-gap-sm': gap?.sm}),
    ...(gap?.md && {'--pc-grid-gap-md': gap?.md}),
    ...(gap?.lg && {'--pc-grid-gap-lg': gap?.lg}),
    ...(gap?.xl && {'--pc-grid-gap-xl': gap?.xl}),
    ...(columns?.xs && {'--pc-grid-columns-xs': columns.xs}),
    ...(columns?.sm && {'--pc-grid-columns-sm': columns.sm}),
    ...(columns?.md && {'--pc-grid-columns-md': columns.md}),
    ...(columns?.lg && {'--pc-grid-columns-lg': columns.lg}),
    ...(columns?.xl && {'--pc-grid-columns-xl': columns.xl}),
```

**thought:** We may need to use this pattern to ensure the CSS var fallbacks you have set in `Grid.scss` apply.

https://user-images.githubusercontent.com/32409546/169569138-dce6ee93-0017-440d-8588-80e553dc4ef0.mov

Reply:
These don't render at all when the value is undefined. Something react does I think